src/content/docs/en/guides/integrations-guide/markdoc.mdx

```diff
@@ -283,45 +288,54 @@ export default defineMarkdocConfig({
+## Custom Markdoc nodes / elements
+
+You may want to render standard Markdown elements, such as paragraphs and bolded text, as Astro components. For this, you can configure a [Markdoc node][markdoc-nodes]. If a given node receives attributes, they will be available as component props.
+
+This example renders blockquotes with a custom `Quote.astro` component:
 
 ```js title="markdoc.config.mjs"
-import { defineMarkdocConfig, nodes } from '@astrojs/markdoc/config';
+import { defineMarkdocConfig, nodes, component } from '@astrojs/markdoc/config';
 
 export default defineMarkdocConfig({
   nodes: {
-    document: {
-      ...nodes.document, // Apply defaults for other options
-      render: null, // default 'article'
+    blockquote: {
+      ...nodes.blockquote, // Apply Markdoc's defaults for other options
+      render: component('./src/components/Quote.astro'),
     },
   },
 });
 ```
 
-### Custom Markdoc nodes / elements
+### Custom headings
 
-You may want to render standard Markdown elements, such as paragraphs and bolded text, as Astro components. For this, you can configure a [Markdoc node][markdoc-nodes]. If a given node receives attributes, they will be available as component props.
+`@astrojs/markdoc` automatically adds anchor links to your headings, and [generates a list of `headings` via the content collections API](/en/guides/content-collections/#rendering-content-to-html). To further customize how headings are rendered, you can apply an Astro component [as a Markdoc node][markdoc-nodes].
 
-This example renders blockquotes with a custom `Quote.astro` component:
+This example renders a `Heading.astro` component using the `render` property:
 
 ```js title="markdoc.config.mjs"
 import { defineMarkdocConfig, nodes, component } from '@astrojs/markdoc/config';
 
 export default defineMarkdocConfig({
   nodes: {
-    blockquote: {
-      ...nodes.blockquote, // Apply Markdoc's defaults for other options
-      render: component('./src/components/Quote.astro'),
+    heading: {
+      ...nodes.heading, // Preserve default anchor link generation
+      render: component('./src/components/Heading.astro'),
     },
   },
 });
 ```
 
-<ReadMore>See the [Markdoc nodes documentation](https://markdoc.dev/docs/nodes#built-in-nodes) to learn about all the built-in nodes and attributes.</ReadMore>
+All Markdown headings will render the `Heading.astro` component and pass the following `attributes` as component props:
 
-### Custom image components in Markdoc
+* `level: number` The heading level 1 - 6
+* `id: string` An `id` generated from the heading's text contents. This corresponds to the `slug` generated by the [content `render()` function](/en/guides/content-collections/#rendering-content-to-html).
+
+For example, the heading `### Level 3 heading!` will pass `level: 3` and `id: 'level-3-heading'` as component props.
+
+### Custom image components
```

Comment:
Just a note for translators: this heading has changed from "Custom image components in Markdoc" to just "Custom image components" It's more than just moved!!  :purple_heart: